### PR TITLE
Handle empty root site collection in a web application.

### PR DESCRIPTION
**Pull request by lavanyachandramouli**
Fix for b/33263458.

### CODE REVIEW

File: src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java [hunk 1]
```diff
@@ -757,7 +757,21 @@ public void init(AdaptorContext context) throws Exception {
...
       }
-      
+      // Web Application which does not have root site is not supported 
+      // by SharePoint. Adaptor returns Error: Document not found (404) 
+      Holder<String> rootSite = new Holder<String>();
+      Holder<String> rootWeb = new Holder<String>();
+      long result = sharePointSiteDataClient.getSiteAndWeb(
+          configuredSharePointUrl.getVirtualServerUrl(), rootSite, rootWeb);
+      if (result != 0) {
+        log.log(Level.WARNING, "Ensure web application has root site created."
```
Comment (wiarlawd) [inline src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java]: There are missing spaces in the message at the line breaks, and no placeholder for the getVirtualServerUrl() argument, but I don't think the extra text here is useful, and so this is a log-and-throw anti-pattern. I would just include the URL in the exception text and drop the logging statement.

File: test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java [hunk 2]
```diff
@@ -362,6 +367,19 @@ public void testInitDestroy() throws Exception {
   }
   
   @Test
+  public void testInitFailedForMissingRootCollection() throws Exception {
+    MockSoapFactory rootMissingSoapFactory = MockSoapFactory.blank().endpoint(VS_ENDPOINT,
```
Comment (wiarlawd) [inline test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java]: 80 cols in a few places here.

File: test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java [hunk 3]
```diff
@@ -381,7 +399,8 @@ public void testAdaptorWithSocketTimeoutConfiguration() throws Exception {
     }
 
     MockSiteData siteDataSoap = new MockSiteData()
-        .register(VS_CONTENT_EXCHANGE).register(CD_CONTENT_EXCHANGE);
+        .register(VS_CONTENT_EXCHANGE).register(CD_CONTENT_EXCHANGE)
```
Comment (wiarlawd) [inline test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java]: Split these onto 3 lines consistently here and below.

File: test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java [hunk 4]
```diff
@@ -1313,6 +1349,7 @@ public void testPolicyAclsWithClaims() throws Exception {
     
     SoapFactory siteDataFactory = MockSoapFactory.blank()
         .endpoint(VS_ENDPOINT, MockSiteData.blank()
+            .register(ROOT_SITE_SAW_EXCHANGE)
```
Comment (wiarlawd) [inline test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java]: Why is the order changing from place to place?

**Review by wiarlawd — CHANGES_REQUESTED** [review]

File: src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java (hunk 1 above)
Comment (lavanyachandramouli) [reply src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java]: Done.

File: test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java (hunk 2 above)
Comment (lavanyachandramouli) [reply test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java]: Done. Thank you.

File: test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java (hunk 3 above)
Comment (lavanyachandramouli) [reply test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java]: Done.

File: test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java (hunk 4 above)
Comment (lavanyachandramouli) [reply test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java]: Done.

**Review by wiarlawd — APPROVED** [review]

**Pull request merged**